075-auto-rebalancing-cluster-scaling.md

```diff
@@ -0,0 +1,368 @@
+# Auto-rebalancing on cluster scaling
+
+This proposal is about enabling the Cluster Operator to rebalance the Apache Kafka cluster automatically when the user scales the cluster up/down by adding/removing brokers to/from the cluster.
+Such a rebalancing can happen right after the scale up to move some partition replicas to the newly added brokers or right before the scale down to move partition replicas off the brokers to be removed.
+
+## Current situation
+
+When a user scales a cluster up by adding new brokers, they would like to use the newly added brokers to host already existing partition replicas and not only for the newly created ones.
+At the same time, in order to scale the cluster down, the brokers to be removed don't have to host partition replicas otherwise the operation is not allowed at all (more details on proposal [049](https://github.com/strimzi/proposals/blob/main/049-prevent-broker-scale-down-if-it-contains-partition-replicas.md)).
+The way the user has to move partition replicas in the above scenarios is via the Cruise Control integration by using the `KafkaRebalance` custom resource.
+
+Currently, there is no automatic rebalancing when the user scales the cluster up or down.
+It means that:
+
+* after scaling the cluster up, in order to rebalance the cluster and move some partition replicas to the newly added brokers, the user has to manually create a `KafkaRebalance` custom resource using the `spec.mode: add-brokers` mode with the new brokers' IDs set on the `spec.brokers` field.
+* before scaling the cluster down, in order to rebalance the cluster and move partition replicas off the brokers to be removed, the user has to manually create a `KafkaRebalance` custom resource using the `spec.mode: remove-brokers` mode with the IDs of the brokers to be removed set on the `spec.brokers` field.
+
+For more details, the proposal [035](https://github.com/strimzi/proposals/blob/main/035-rebalance-types-scaling-brokers.md) introduced the rebalancing modes mentioned above, which will be also referenced across this proposal.
+
+## Motivation
+
+The rebalancing operation during scaling up and down is a manual two steps process.
+
+* scaling the cluster up by adding brokers, then running a rebalance.
+* running a rebalance, then scaling the cluster down by removing brokers.
+
+In order to simplify the process, it would be great for the user to deal with just scaling the cluster and then having the operator taking care of running a rebalancing operation automatically when needed.
+
+## Proposal
+
+This proposal is about providing the user the possibility to automatically run a rebalancing when they scale the cluster up (by adding brokers) or scale the cluster down (by removing brokers) by changing the `spec.replicas` within a `Kafka` or `KafkaNodePool` custom resource.
```

Comment:
Is it worth mentioning that it does not apply for node pool deletion? (Only scale-down)

Reply:
@scholzj I was thinking the same. When you delete or add a node pool, at Kafka reconciler level doesn't you get anyway `KafkaCluster.addedNodes()` and `KafkaCluster.removeNodes()` filled with the brokers from the pools? Or deleting/adding a pool will need a special care?